packages/next/build/index.ts

```diff
@@ -2151,6 +2155,13 @@ export default async function build(
                 }
               } else {
                 hasDynamicData = true
+                // we might have determined during prerendering that this page
+                // used dynamic data
+                pageInfos.set(route, {
```

Comment:
this is necessary because we can only if a route is dynamic after trying to render it now